addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ScriptsListPanel.java

```diff
@@ -690,10 +690,16 @@ public void showInTree(ScriptNode node) {
     }
 
     public void showInTree(ScriptNode node, boolean expand) {
+        showInTree(node, expand, true);
+    }
+
+    public void showInTree(ScriptNode node, boolean expand, boolean allowFocus) {
         TreeNode[] path = node.getPath();
         TreePath tp = new TreePath(path);
         getTree().setExpandsSelectedPaths(true);
-        getTree().setSelectionPath(tp);
+        if (allowFocus) {
```

Comment:
Why not allow the path selection with focusing? Doesn't that work?

Reply:
I have updated the code. Please check whether we can do it this way or not

Reply:
Because I `setSelectionPath` internally invokes displayScript function which is responsible for setting the focus. We we create it inside ScriptsListPanel we will need two functions to set and read the value. Should I do it that way?

Reply:
Can't we pass the `allowFocus`, like talked in https://github.com/zaproxy/zap-extensions/pull/4809#discussion_r1313832609 ?

Reply:
ohh got it. Yes we can do that.